src/core/tests/type_prop/string_tensor_pack.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/op/string_tensor_pack.hpp"
+
+#include <gtest/gtest.h>
+
+#include "common_test_utils/test_assertions.hpp"
+#include "common_test_utils/type_prop.hpp"
+
+namespace ov {
+namespace test {
+
+using ov::op::v0::Parameter;
+using testing::HasSubstr;
+
+class TypePropStringTensorPackTestSuite : public ::testing::TestWithParam<ov::PartialShape> {};
+
+TEST_P(TypePropStringTensorPackTestSuite, TypePropStringTensorPackTestSuite) {
+    const auto& indices_shape = GetParam();
+    const auto begins = std::make_shared<Parameter>(element::i32, indices_shape);
+    const auto ends = std::make_shared<Parameter>(element::i32, indices_shape);
```

Comment:
Is possible to have one input dynamic() and other e.g dynamic(2) or {3}? Add such test scenarios.

Reply:
Added.

Reply:
It's possible, but in this case currently the output shape is always `begins_shape`. The way it should work is to pick the more "strict" of the shapes, including intervals support. Since I can't use `PartialShape::merge_into` due to one of the TShapes being from CPU (and I'm not sure it works for 2 dynamic shapes as needed here), I would prefer to add this dynamic output shape improvement in the next PR - reference implementation. 

This one already has over 1000 lines changed, the reference implementation will be rather short and this improvement will be a lot of manual Dimension checking. Please let me know if you'd rather have the improvement in the current PR, I'll add it then.